distributed/shuffle/shuffle_extension.py

```diff
@@ -27,93 +25,62 @@
 ShuffleId = NewType("ShuffleId", str)
 
 
-# NOTE: we use these dataclasses primarily for type-checking benefits.
-# They take the place of positional arguments to `shuffle_init`,
-# which the type-checker can't validate when it's called as an RPC.
-
-
-@dataclass(frozen=True, eq=False)
-class NewShuffleMetadata:
-    "Metadata to create a shuffle"
-    id: ShuffleId
-    empty: pd.DataFrame
-    column: str
-    npartitions: int
-
-
-@dataclass(frozen=True, eq=False)
-class ShuffleMetadata(NewShuffleMetadata):
-    """
-    Metadata every worker needs to share about a shuffle.
-
-    A `ShuffleMetadata` is created with a task and sent to all workers
-    over the `ShuffleWorkerExtension.shuffle_init` RPC.
-    """
-
-    workers: list[str]
-
-    def worker_for(self, output_partition: int) -> str:
-        "Get the address of the worker which should hold this output partition number"
-        assert output_partition >= 0, f"Negative output partition: {output_partition}"
-        if output_partition >= self.npartitions:
-            raise IndexError(
-                f"Output partition {output_partition} does not exist in a shuffle producing {self.npartitions} partitions"
-            )
-        return worker_for(output_partition, self.workers, self.npartitions)
-
-    def _partition_range(self, worker: str) -> tuple[int, int]:
-        "Get the output partition numbers (inclusive) that a worker will hold"
-        i = self.workers.index(worker)
-        first = math.ceil(self.npartitions * i / len(self.workers))
-        last = math.ceil(self.npartitions * (i + 1) / len(self.workers)) - 1
-        return first, last
-
-    def npartitions_for(self, worker: str) -> int:
-        "Get the number of output partitions a worker will hold"
-        first, last = self._partition_range(worker)
-        return last - first + 1
-
-
```

Comment:
My condolences MetaData object.  May you rest in peace.  

Things changed enough from the original design that keeping this around was, i think, more of a hindrance and an artificial abstraction than helpful.